xapi-client/src/test/java/dev/learning/xapi/client/XapiClientMultipartTests.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2016rue-2023 Berry Cloud Ltd. All rights reserved.
+ */
+package dev.learning.xapi.client;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+import dev.learning.xapi.model.Activity;
+import dev.learning.xapi.model.Agent;
+import dev.learning.xapi.model.Statement;
+import dev.learning.xapi.model.SubStatement;
+import dev.learning.xapi.model.Verb;
+import java.net.URI;
+import java.util.Locale;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.web.reactive.function.client.WebClient;
+
+/**
+ * XapiClient Tests.
+ *
+ * @author Thomas Turrell-Croft
+ */
+@DisplayName("XapiClient Tests")
+@SpringBootTest
+class XapiClientMultipartTests {
+
+  @Autowired
+  private WebClient.Builder webClientBuilder;
+
+  private MockWebServer mockWebServer;
+  private XapiClient client;
```

Comment:
Auto-configuration?

Reply:
We can't use autoconfiguration here. we create a new mock for each test, so  we nned a new client woth the appropriate url each time.